src/DIRAC/WorkloadManagementSystem/Agent/JobAgent.py

```diff
@@ -545,28 +546,28 @@ def _matchAJob(self, ceDictList):
                 break
         return jobRequest
 
-    def _checkMatchingIssues(self, issueMessage):
+    def _checkMatchingIssues(self, jobRequest):
         """Check the source of the matching issue
 
-        :param str issueMessage: message returned by the matcher
+        :param dict jobRequest: S_ERROR returned by the matcher
         :return: S_OK/S_ERROR
         """
-        if issueMessage.find("Pilot version does not match") != -1:
+
+        if jobRequest["Message"].find("Pilot version does not match") != -1:
             errorMsg = "Pilot version does not match the production version"
-            self.log.error(errorMsg, issueMessage.replace(errorMsg, ""))
-            return S_ERROR(issueMessage)
+            self.log.error(errorMsg, jobRequest["Message"].replace(errorMsg, ""))
```

Comment:
I think returning S_ERROR here was on purpose: there is no sense to continue requesting jobs if the pilot version is not accepted